Architecture Diagrams (#201)
mulesoft/docs-gateway: gateway/1.4/modules/ROOT/pages/flex-architecture-basic-deployments.adoc

= Basic Deployment Models for Flex Gateway
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:imagesdir: ../assets/images

> Flex Gateway supports three deployment models, all of which apply to Connected Mode and Local Mode.

Reviewer: ```suggestion
Flex Gateway supports three basic deployment models, all of which apply to Connected Mode and Local Mode.
```

> 

Reviewer: General suggestion: For cases like this where you have a lead-in and then show the options as headings, I like to do a list here to anchor the user. Each list option would be a link to the appropriate section. For example:

Flex Gateway supports three basic deployment models, all of which apply to Connected Mode and Local Mode:

* Standalone
* Ingress
* Sidecar

Reviewer: Suggestion:
Flex Gateway supports three basic deployment models, which apply to both Connected Mode and Local Mode.

> NOTE: This overview uses Kubernetes terminology, but you can extend to other technological stacks with the appropriate network controls applied.

Reviewer: ```suggestion
NOTE: This overview uses Kubernetes terminology, but you can extend the overview to other technological stacks that have the appropriate network controls applied.
```

Reviewer: I don't think you need a NOTE: here. Having a separate paragraph is enough.

== Standalone Deployment

> In standalone deployment, Flex Gateway acts as a standalone service to protect one or more integration API flows to manage internal traffic. You deploy Flex Gateway in a separate pod and namespace from the consumer applications.

Reviewer: ```suggestion
In a standalone deployment, Flex Gateway acts as a standalone service that protects one or more integration API flows by managing internal traffic. In this type of deployment, you deploy Flex Gateway in a separate pod and namespace from the consumer applications:
```

Reviewer: Not sure if my wording above is correct. What you have is unclear.

> image:ad-standalone-deployment.png[In standalone deployment, Flex Gateway and the consumer applications are in different Kubernetes namespaces.]

Reviewer: ```suggestion
image:ad-standalone-deployment.png[In a standalone deployment, Flex Gateway and the consumer applications are in different Kubernetes namespaces.]
```

> In the diagram, all external traffic passes through a load balancer before entering the Flex Gateway namespace. The traffic then passes through Flex Gateway before reaching the consumer application namespace.

Reviewer: ```suggestion
As the diagram shows, all external traffic in a standalone deployment passes through a load balancer before entering the Flex Gateway namespace. The traffic then passes through Flex Gateway before reaching the consumer application namespace.
```

== Ingress Deployment

> In ingress deployment, Flex Gateway acts as a standalone service that protects applications by managing incoming request. You deploy Flex Gateway in a separate pod and namespace from the consumer applications. Ingress deployment is similar to standalone deployment, but traffic in ingress deployment comes from external services.

Reviewer: ```suggestion
In an ingress deployment, Flex Gateway acts as a standalone service that protects applications by managing incoming requests. In this type of deployment, you deploy Flex Gateway in a separate pod and namespace from the consumer applications. An ingress deployment is similar to a standalone deployment, but traffic in an ingress deployment comes from external services.
```

image:ad-ingress-deployment.png[[In ingress deployment, Flex Gateway and the consumer applications are in different Kubernetes namespace. Flex Gateway is used as an ingress controller.]

> In the diagram, all external traffic passes through a load balancer before entering the Flex Gateway namespace. The traffic then passes through Flex Gateway, before reaching the consumer application namespace.

Reviewer: ```suggestion
As the diagram shows, all external traffic in an ingress deployment passes through a load balancer before entering the Flex Gateway namespace. The traffic then passes through Flex Gateway before reaching the consumer application namespace.
```

== Sidecar Deployment

> In sidecar deployment, you deploy Flex Gateway in the same pod as the service it is protecting. Each Flex Gateway deployment exclusively protects the APIs exposed by its protected service. A new Flex Gateway replica is added with each new protected service. 

Reviewer: ```suggestion
In a sidecar deployment, you deploy Flex Gateway in the same pod as the service that it is protecting. Each Flex Gateway deployment exclusively protects the APIs exposed by its protected service. A new Flex Gateway replica is added with each new protected service. 
```

image:ad-sidecar-deployment.png[In sidecar deployment, Flex Gateway and the consumer application are in the same Kubernetes pod.]

> In the diagram, traffic passes from the load balancer, to Flex Gateway, then the consumer application. Flex Gateway and the consumer application are both in the same pod.

Reviewer: ```suggestion
As the diagram shows, traffic in a sidecar deployment passes from the load balancer to Flex Gateway, and then to the consumer application. Flex Gateway and the consumer application are both in the same pod.
```

Reviewer: Should "traffic" be "external traffic" as in the other types of deployments?

== See Also

> * xref:flex-architecture-basic-deployments.adoc[]

Reviewer: The link text is not showing in the See Also in the preview. Is that because it refers to a different repository?

= Basic Deployment Models for Flex Gateway
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:imagesdir: ../assets/images

> Flex Gateway supports three basic deployment models, which apply to both Connected Mode and Local Mode:

Reviewer: Should Connected Mode and Local Mode be lowercase?

Author: No, we capitalized these across the docs. Maybe we should change it though.

> * <<standalone-deployment, Standalone Deployment>>

Reviewer: "Deployment" should be lowercase in these entries because they are in context. (I can explain to you verbally if needed.)

* <<ingress-deployment, Ingress Deployment>>
* <<sidecar-deployment, Sidecar Deployment>>

The deployment models in this overview use Kubernetes terminology, but you can extend the deployment models to other technological stacks that have the appropriate network controls applied.

== Standalone Deployment

In a standalone deployment, Flex Gateway acts as a standalone service that protects one or more integration API flows by managing internal traffic. In this type of deployment, you deploy Flex Gateway in a separate pod and namespace from the consumer applications:

> image:ad-standalone-deployment.png[In a standalone deployment, Flex Gateway and the consumer applications are in different Kubernetes namespaces.]

Reviewer: I forgot to add this global edit - all alt text that has spaces or punctuation must be enclosed in quotes. Gary says that it doesn't render correctly otherwise.

As the diagram shows, all external traffic in a standalone deployment passes through a load balancer before entering the Flex Gateway namespace. The traffic then passes through Flex Gateway before reaching the consumer application namespace.

== Ingress Deployment

In an ingress deployment, Flex Gateway acts as an ingress controller that protects applications by managing incoming requests. In this type of deployment, you deploy Flex Gateway in a separate pod and namespace from the consumer applications. An ingress deployment is similar to a standalone deployment, but traffic in an ingress deployment comes from external services:

image:ad-ingress-deployment.png[[In an ingress deployment, Flex Gateway and the consumer applications are in different Kubernetes namespace. Flex Gateway acts as an ingress controller.]

As the diagram shows, all external traffic in an ingress deployment passes through a load balancer before entering the Flex Gateway namespace. The traffic then passes through Flex Gateway before reaching the consumer application namespace.

== Sidecar Deployment

> In a sidecar deployment, you deploy Flex Gateway in the same pod as the service that it is protecting. Each Flex Gateway deployment exclusively protects the APIs exposed by its protected service. A new Flex Gateway replica is added with each new protected service:

Reviewer: ```suggestion
In a sidecar deployment, you deploy Flex Gateway in the same pod as the service that it is protects. Each Flex Gateway deployment exclusively protects the APIs exposed by its protected service. A new Flex Gateway replica is added with each new protected service:
```

image:ad-sidecar-deployment.png[In a sidecar deployment, Flex Gateway and the consumer application are in the same Kubernetes pod.]

> As the diagram shows, external traffic in a sidecar deployment passes from the load balancer to Flex Gateway, and then to the consumer application. Flex Gateway and the consumer application are both in the same pod.

Reviewer: ```suggestion
As the diagram shows, external traffic in a sidecar deployment passes from the load balancer to Flex Gateway, and then to the consumer application. Flex Gateway and the consumer application are in the same pod.
```
